### PR TITLE
[SIL Opaque Value] Address Lowering: Add support for unconditional_checked_cast

### PR DESCRIPTION
**Pull request by meg-gupta**

### CODE REVIEW

**Comment by meg-gupta** [comment]
@swift-ci test

File: lib/SILOptimizer/Mandatory/AddressLowering.cpp [hunk 1]
```diff
@@ -2979,6 +3011,31 @@ class DefRewriter : SILInstructionVisitor<DefRewriter> {
     assert(!storage.storageAddress || storage.storageAddress == addr);
     storage.storageAddress = addr;
   }
+
+  void visitUnconditionalCheckedCastInst(
+      UnconditionalCheckedCastInst *uncondCheckedCast) {
+    SILValue srcVal = uncondCheckedCast->getOperand();
+    assert(srcVal->getType().isLoadable(*pass.function));
+    // Create a stack temporary to store the srcVal
+    SILBuilderWithScope allocBuilder(uncondCheckedCast);
```
Comment (atrick) [inline lib/SILOptimizer/Mandatory/AddressLowering.cpp:3020]: I don't think you actually need to create any builders in this function. Everything is emitted at the same scope/loc.

**Review by atrick — APPROVED** [review]
Thanks! I have some code simplification requests.

File: lib/SILOptimizer/Mandatory/AddressLowering.cpp [hunk 2]
```diff
@@ -2560,6 +2560,38 @@ class UseRewriter : SILInstructionVisitor<UseRewriter> {
...
+    SILValue srcVal = uncondCheckedCast->getOperand();
+    assert(srcVal->getType().isAddressOnly(*pass.function));
+    SILValue srcAddr = pass.valueStorageMap.getStorage(srcVal).storageAddress;
+
+    // Cast destination maybe a loadable type or address only type
+    SILValue destAddr;
+    if (uncondCheckedCast->getType().isLoadable(*pass.function)) {
+      // For loadable cast destination type, create a stack temporary
+      SILBuilderWithScope allocBuilder(uncondCheckedCast);
+      destAddr = allocBuilder.createAllocStack(uncondCheckedCast->getLoc(),
```
[the remark this answers is not in the document]
Comment (meg-gupta) [reply lib/SILOptimizer/Mandatory/AddressLowering.cpp]: Thanks. I updated with `pass.getBuilder()`

**Comment by meg-gupta** [comment]
@swift-ci test and merge

**Pull request merged**